gobblin-metastore/src/main/java/org/apache/gobblin/metastore/MysqlJobStatusStateStore.java

```diff
@@ -70,6 +71,21 @@ public MysqlJobStatusStateStore(DataSource dataSource, String stateStoreTableNam
     return getAll(storeName, flowExecutionId + "%", true);
   }
 
+  @Override
+  protected String getCreateJobStateTableTemplate() {
```

Comment:
Would be good to add a couple of tests that will verify the state store behavior with long names. Older version of MySql with a certain db layout types would limit the total length of the primary key to about 700-800 bytes. At the same time, it was possible to define any limits in table schema, even more than 800 total bytes. But, when you try to insert a long key, a runtime exception will occur.

Reply:
Ok, so what should a test verify?

Reply:
The test can write the state for the flow with max length, and max expected table length, and then try to read it.

Reply:
FYI, I added a test in other module to test long names when creating spec with invalid names.